Update spec contrib guidelines in Elasticsearch API docs quickstart (#7461)
elastic/docs-content: contribute-docs/api-docs/elasticsearch-api-docs-quickstart.md

::::{step} Make your docs changes
Edit the relevant TypeScript files in the `specification` directory. Use JSDoc comments to describe your API interfaces, following the [guidelines](./guidelines.md). Add or update summaries, descriptions, tags, metadata, links, and examples as needed.
::::

::::{step} Format your changes
```shell
make spec-format-fix
```
Run this command to format your changes according to the standard used in the specification.
::::

::::{step} Open a pull request

Once you're satisfied with your docs changes:
1. Create a pull request from a branch on your local clone

> 2. Make sure not to include any generated output file in the PR, otherwise the backporting job will likely fail

Reviewer: The rule would be safer if it named the generated paths contributors are likely to see after running the preview commands.

```suggestion
2. Do not include generated files in the PR, i.e. files under `output/` and `docs/examples/languageExamples.json`. CI regenerates these files, and committing them can cause backport jobs to fail.
```

3. The CI will generate the output and validate your additions
4. Once approved, merge your changes and ensure they are backported to the appropriate branches
::::

:::::

This is everything that is needed to contribute to the spec, as most validation is performed by CI. 
If you need to preview the changes locally and check how they'll be displayed in Bump, then a few more steps are needed.

> ## Preview Bump changes locally

Reviewer: Because Bump hosts the resulting preview remotely, this title is a little more precise.

```suggestion
## Generate a Bump preview
```

```

> ::::

Reviewer: The optional preview path recreates the files prohibited above, so it would help to repeat the warning at the point where they appear.

```suggestion
After generating the previews, make sure files under `output/` and `docs/examples/languageExamples.json` are not included in your pull request. CI regenerates these files.
::::
```
